pkg/controllers/vdb/sandboxsubcluster_reconciler.go

```diff
@@ -68,9 +72,51 @@ func (s *SandboxSubclusterReconciler) Reconcile(ctx context.Context, _ *ctrl.Req
 		return ctrl.Result{}, err
 	}
 
+	// reconcile sandbox status for the subclusters that are already sandboxed
+	if err := s.reconcileSandboxStatus(ctx); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// reconcile sandbox config maps for the existing sandboxes
+	if err := s.reconcileSandboxConfigMaps(ctx); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	return s.sandboxSubclusters(ctx)
 }
 
+// reconcileSandboxStatus will update sandbox status for the subclusters that are already sandboxed
+func (s *SandboxSubclusterReconciler) reconcileSandboxStatus(ctx context.Context) error {
+	sbScMap := make(map[string][]string)
+	seenScs := make(map[string]any)
+	for _, v := range s.PFacts.Detail {
+		if _, ok := seenScs[v.subclusterName]; ok {
+			continue
+		}
+		if v.sandbox != vapi.MainCluster {
```

Comment:
I expected there would be a change to podfacts.go. It looks like we set v.sandbox initially in `collectPodByStsIndex` and we get it from the requested sandbox we are requesting. That doesn't seem right. I feel like we should pull it from a label in the statefulset. The comment seems to indicate that, but I'm not sure if that's exactly what's going on.

This is the code I am talking about:
```
		// we get the sandbox name from the sts labels if the subcluster
		// belongs to a sandbox. If the node is up, we will later retrieve
		// the sandbox state from the catalog
		pf.sandbox = p.SandboxName
```

`p.SandboxName` is just an input parameter. I wonder if it should be `sts.Labels[vmeta.SandboxNameLabel`.

Reply:
I will change it to sts.Labels[vmeta.SandboxNameLabel] since it probably causes an error when the node is down. That time we cannot update it to the correct sandbox name from checkNodeDetails. Then it could make the node restart fail.